highfidelity/hifi
# 20116 nitpick v1.2

https://highfidelity.fogbugz.com/f/cases/20116/Nitpick-v1-2
# Description
1.  The HTML test result file created on Mac is now sorted correctly.  This is done by iterating over the folder, reading entries into a string list, and then sorting.
2.  After an HTML file is created, a notification message appears.  This was missing.
3.  Some folders have no automated tests in them, but nitpick was still creating recursive scripts for the parent folders.  This was creating empty recursive scripts.
4.  Protocol tests now produce a text file (JSON) as a result.  Nitpick has been modified to read these files as part of the test evaluation.
# Testing
Use of nitpick is documented in: https://github.com/highfidelity/hifi/blob/master/tools/nitpick/README.md
================================================
1.  On Mac: download installer from https://hifi-qa.s3.amazonaws.com/nitpick/Mac/nitpick-installer-v1.2.dmg
2.  Double-click on the downloaded image to mount it
3. Create a folder for the nitpick files (e.g. ~/nitpick) If this folder exists then delete all it's contents.
4. Copy the downloaded files to the folder
    In a terminal:
cd ~/nitpick
    cp -r /Volumes/nitpick-installer-v1.2/* .

5. Run nitpick - _cd_ to the folder that you copied to, and enter ./nitpick
6.  Download the following zipped results folder: https://hifi-qa.s3.amazonaws.com/nitpick/20116/Zipped%20Results/TestResults--2018-12-05_15-04-49%28local%29%5BDESKTOP-6BO62Q9%5D.zip
7.  Use this file to create an HTML results file.
8.  Verify that a "HTML file message pops-up.  Close this message.
9.  Verify that this file looks like the following (open both in a browser): https://hifi-qa.s3.amazonaws.com/nitpick/20116/HTML%20Results/TestResults--2018-12-05_15-04-49%28local%29%5BDESKTOP-6BO62Q9%5D.zip
=========================================================
10.  On Windows:  download installer from https://hifi-qa.s3.amazonaws.com/nitpick/Windows/nitpick-installer-v1.2.exe
11.  Install and run nitpick
12.  Clone the hifi_tests repository from https://github.com/highfidelity/hifi_tests
13.  Count the number of testRecursive.js files in the hifi_tests\tests directory (e.g. using the Explorer search field).
14.  Use nitpick to create all recursive scripts.
15.  Verify that the number of testRecursive.js scripts has not increased (it may decrease - as the repository may contain previous empty recursive scripts.).
